ClosedXML/Excel/CalcEngine/CalcEngineHelpers.cs

```diff
@@ -54,11 +59,26 @@ internal static bool ValueSatisfiesCriteria(object value, object criteria, CalcE
                 }
 
                 // if criteria is a regular expression, use regex
-                if (cs.IndexOf('*') > -1)
+                if (cs.IndexOfAny(new[] { '*', '?' }) > -1)
                 {
-                    var pattern = cs.Replace(@"\", @"\\");
-                    pattern = pattern.Replace(".", @"\");
-                    pattern = pattern.Replace("*", ".*");
+                    var patternReplacements = new Dictionary<string, Tuple<string, string>>();
```

Comment:
I think this should rather be a private static field in the class. Populate it in a static constructor.

Reply:
I'm not obsessed with my solution here, so I can change it of course, but I would go for a compromise:
We're talking about a single dictionary, so it's not that much, but keeping a static dictionary in memory is wasting a little bit, but heap space, that's only necessary if there's any string pattern function actually being evaluated.
So I would go for a static, but lazy initialized field.

Reply:
Yeah, I don't want a dictionary to be populated every time a cell formula is evaluated. I'll sacrifice the slight extra memory for the performance improvement.